### PR TITLE
sql: add owners to various pg_catalog tables

### LINKED ISSUE #53491: sql: populate pg_catalog for owners
identify and populate owners in virtual tables

### PR DESCRIPTION
**Pull request by rohany**
Fixes #53491.

Release justification: low-risk, high benefit changes to existing
functionality

Release note (sql change): The related `owner` columns are now populated
in `pg_catalog` metadata tables.

### CODE REVIEW

**Review by solongordon — APPROVED** [review]
Good stuff, thanks for knocking this out!

As a side note I think we should also consider adding an owner column to SHOW TABLES, SHOW DATABASES, etc. but that's outside the scope of this PR.

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53495#-:-MFkSETj04OU92B6cC36:b-tydg0r)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

File: pkg/sql/authorization.go [hunk 1]
```diff
@@ -138,11 +138,9 @@ func (p *planner) CheckPrivilege(
 	return p.CheckPrivilegeForUser(ctx, descriptor, privilege, p.User())
 }
 
-// IsOwner returns if the role has ownership on the descriptor.
-func IsOwner(desc catalog.Descriptor, role string) bool {
+func getOwnerOfDesc(desc catalog.Descriptor) string {
```
Comment (otan) [inline pkg/sql/authorization.go:141]: wondering if this should be a property of Descriptor. Or named `descriptorOwner(desc)`.
this is a very minor nit and i don't mind either way - i just know [`Get` is kind of a frowned word in Go](https://golang.org/doc/effective_go.html#Getters).

**Review by otan — APPROVED** [review]

File: pkg/sql/authorization.go (hunk 1 above)
Comment (rohany) [reply pkg/sql/authorization.go:141]: This is a good point, but I think should be done when we push out this authorization stuff from `sql`.

**Comment by rohany** [comment]
> As a side note I think we should also consider adding an owner column to SHOW TABLES, SHOW DATABASES, etc. but that's outside the scope of this PR.

Yeah, that should be done. I can make issues for that.

bors r+

**Pull request merged**